daffodil-test/src/test/resources/org/apache/daffodil/section06/namespaces/namespaces.tdml

```diff
@@ -2415,4 +2417,23 @@
     </tdml:infoset>
   </tdml:parserTestCase>
 
+  <!--
+    Test name: schemaNoGlobalElems_03
+    Schema: multi_base_21.dfdl.xsd
+    Root: base2
+    Purpose: This test demonstrates that you can have a schema if there are
+             no global elements in the primary schema, as long as the root element is
+             specified (the TDML Runner always specifies the root)
```

Comment:
I searched TDMLRunner.scala for "root" to understand what you meant by "the TDML Runner always specifies the root".  What seems to happen is if a TDML test has an infoset, the TDML Runner uses that infoset's root element (XML documents have a single root element) as the root.  The root="name" attribute is not even needed and in fact the TDML Runner throws an exception if the root="name" attribute does not match the name of the root element of the infoset.

Reply:
Yeah, the root attribute is technically redundant. I don't think that was always the case, not sure when that feature was added. I wonder if we should have have a daffodil standard that the root attribute shouldn't be provided for positive test cases? I do kindof like it just for the sanity check. I know I sometimes copy the infoset from another test and forget to update the root element. In that case I get an error about the root attribute not matching.